fedimint-server/src/lib.rs

```diff
@@ -193,20 +203,40 @@ impl FedimintServer {
         module: RpcModule<RpcHandlerCtx<T>>,
         max_connections: u32,
         task_group: &mut TaskGroup,
+        force_shutdown: bool,
     ) {
-        let runtime = Runtime::new().expect("Creates runtime");
-        let handle = ServerBuilder::new()
-            .max_connections(max_connections)
-            .custom_tokio_runtime(runtime.handle().clone())
-            .ping_interval(Duration::from_secs(10))
-            .build(&self.settings.api_bind.to_string())
-            .await
-            .context(format!("Bind address: {}", self.settings.api_bind))
-            .expect("Could not start API server")
-            .start(module)
-            .expect("Could not start API server");
-
-        let handler = FedimintApiHandler { runtime, handle };
+        // TODO: Remove this flag needed only for tests
+        let handler = if force_shutdown {
+            let runtime = Runtime::new().expect("Creates runtime");
```

Comment:
If it doesn't work in tests, how is it going to work in prod? :(

Reply:
Well it works in the test I have against the `test_config_api` test, but there's something about running it in the rust-tests.sh that is causing issues.

Reply:
I'm not a fan of that workaround, and would rather rely on a proper shutdown detection, but if it unblocks you, let's land. Maybe add some `// Workaround:` comment, link to the discussion and issues` and in the future, once the original root cause is fixed, we might revert that part.

Reply:
Also - maybe let's fill a `jsonrpcsee` issue to supported forced shutdown? It should be too difficult to implement at this level, and it seems very useful to have.

Reply:
Okay just pushed a much better solution without a flag.  Instead I return the handle and the caller deals with it.  I think the callback in the `TaskGroup` must be called by one of the many things in the integration test framework before we are ready to shutdown the API.

Reply:
I see. So instead of trying to guarantee that server is gone when returning, we just return a handle, and now it's a caller's problem to wait for things to finish?

Reply:
Yeah, I like that better.